python-package/xgboost/compat.py

```diff
@@ -88,6 +88,18 @@ def is_cudf_available() -> bool:
         return False
 
 
+def is_cupy_available() -> bool:
+    """Check cupy package available or not"""
+    if importlib.util.find_spec("cupy") is None:
+        return False
+    try:
+        import cupy
+
+        return True
+    except ImportError:
+        return False
```

Comment:
The test for cuDF availability originally had the comment:
```
    # Checking by `importing` instead of check `importlib.util.find_spec("cudf") is not None`
WeichenXu123 marked this conversation as resolved.
    # because user might install cudf successfully but importing cudf raises issues (e.g. saying
    # running on mismatched cuda version)
```
which to me was weird, that's a mismanaged environment and I'm not sure it's necessary for xgboost to work around it (or even a good idea to workaround anything since users with cuDF installed might expect GPU to be used).